application/app/Policies/ProductCategoryPolicy.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace App\Policies;
+
+use App\Models\AdminUser;
+use App\Models\ProductCategory;
+use Illuminate\Auth\Access\HandlesAuthorization;
+
+class ProductCategoryPolicy
+{
+    use HandlesAuthorization;
+
+    /**
+     * @param AdminUser $admin_user
+     * @param ProductCategory $category
+     * @return bool
+     */
+    public function delete(AdminUser $admin_user, ProductCategory $category)
+    {
+        return $category->products->isEmpty();
```

Comment:
```suggestion
        return $category->products()->doesntExist();
```
動的プロパティをつかうとSQLが発行されカラムが取得されます。
今回はカラムが取得したいわけではないので無駄な処理ということになります。

Reply:
なるほど。